features/Pim/Behat/Decorator/Completeness/PanelDecorator.php

```
@@ -0,0 +1,78 @@
+<?php
+
+namespace Pim\Behat\Decorator\Completeness;
+
+use Behat\Mink\Element\NodeElement;
+use Context\Spin\SpinCapableTrait;
+use Pim\Behat\Decorator\ElementDecorator;
+
+class PanelDecorator extends ElementDecorator
+{
+    use SpinCapableTrait;
+
+    /** @var array Selectors to ease find */
+    protected $selectors = [
+        'Completeness blocks' => [
+            'css'        => '.completeness-block',
+            'decorators' => [
+                'Pim\Behat\Decorator\Completeness\BlockDecorator'
+            ]
+        ]
+    ];
+
+    /**
+     * @param string $locale en_US, fr_FR, etc.
+     *
+     * @return null|NodeElement
+     */
+    public function findCompletenessForLocale($locale)
+    {
+        $block = $this->findBlock(
+            sprintf($this->selectors['Completeness blocks']['css'] . ' header .locale[data-locale="%s"]', $locale)
+        );
+        $completeness = $block->getParent()->getParent();
+
+        return $this->decorate($completeness, $this->selectors['Completeness blocks']['decorators']);
+    }
+
+    /**
+     * @param int $position Begin to 1
+     *
+     * @return null|NodeElement
+     */
+    public function findNthCompleteness($position)
+    {
+        $blocks = $this->findAllBlocks($this->selectors['Completeness blocks']['css']);
+        if (!is_array($blocks) || count($blocks) < $position) {
+            return null;
```

should throw an exception